autoload/clap.vim

```diff
@@ -182,12 +182,8 @@ endfunction
 
 " Sometimes we don't need to go back to the start window, hence clap#_exit() is extracted.
 function! clap#exit() abort
+  call g:clap.start.goto_win()
```

Comment:
```suggestion
  noautocmd call g:clap.start.goto_win()
```

How about using `noautocmd` here? Feels like we never leave the start window...

Reply:
No, that's precisely the moment when the autocommands will be triggered. It's because we do it without autocmd that other plugins can update their state correctly. We could add it here, but then we'd need to add `doautocmd WinEnter`. Doing `doautocmd WinEnter` is also only partially correct, because plugins (or the user config) might be listening to other autocmds (`BufEnter`). Letting autocommands be triggered natively is also more future-proof.

One alternative that would really feel to plugins as if the user never leaves the start window is to put `noautocmd` everywhere: from when clap opens, until when clap exits. This would also have the advantage of being more performant because no other scripts can run. However, this is a more involved modification and I don't feel ready to do it at this point.

Reply:
Ok, that makes sense.